clientconn.go

```diff
@@ -234,6 +246,7 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 		case sc, ok := <-cc.dopts.scChan:
 			if ok {
 				cc.sc = &sc
+				cc.safeConfigSelector.UpdateConfigSelector(&defaultConfigSelector{cc})
```

Comment:
Does it make more sense to initialize the `safeConfigSelector` field as part of the literal struct initialization of `ClientConn`, instead of doing it here and in line 298?

Reply:
The whole "service config channel" path annoys me.  If someone pushes something to this channel, I think we need to revert back to the default config selector if a resolver had already pushed a config selector.  But nobody should be using a service config channel with xds, so it also probably doesn't matter.

But I think the whole "service config channel" may actually be somewhat broken now, and we should probably try to delete it soon.

Reply:
Ack